docs/_source/docs/architecture.rst

```diff
@@ -135,15 +135,222 @@ to demonstrate ways of organising different projects.
+            application.yaml    # Stack Config for Indigo's application stack
+        project/                # StackGroup for resources needed to deploy Indigo
+            deployment.yaml     # Stack Config for Indigo's deployment resources
+    templates/
+        indigo/
+            application.yaml
+        project/
+            deployment.yaml
+    vars/
+        production.yaml         # var-file with the production environment's configurations
+        development.yaml        # var-file with configurations for development environments
+
+Example Configs
+***************
+
+Top-level project configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+
+    # config/config.yaml
+
+    # This is the top-level config file for the whole project. Every stack in the project will
+    # inherit these settings unless they explicitly override them. At minimum, you need the
+    # project_code and region, but you can set whatever StackGroup and Stack configurations you
+    # want inherited by the rest of the project.
+
+    # By making the project-code parameterized by environment name, it guarantees that one
+    # environment's stacks do not conflict with another environment's stacks.
+    project_code: indigo-{{ var.environment }}
+    region: {{ var.region }}
+
+    # Setting the template_bucket_name with a !stack_output means that every stack in the project
+    # (except the deployment stack) will have its template uploaded to that bucket. It ALSO means
+    # that every stack in the project will inherit a dependency on the deployment stack (except the
+    # deployment stack... see its comments below)
+    template_bucket_name: !stack_output project/deployment.yaml::BucketName
+
+    # By making the template_key_prefix parameterized by environment, it means that every
+    # environment's templates will be uploaded to its own directory structure within the template
+    # bucket, preventing conflicts.
+    template_key_prefix: {{ var.environment }}
+
+    # Setting the iam_role on the project like this means that all Sceptre actions on all stacks in
+    # the project will be done by assuming the deployment role whose ARN is output by the deployment
+    # stack (except for actions on the deployment stack... see its comments below).
+    iam_role: !stack_output project/deployment.yaml::DeploymentRoleArn
+
+
+The Deployment Stack
+^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+
+    # config/project/deployment.yaml
+
+    # This is a StackConfig for a stack that defines the project's basic requirements, that all
+    # stacks and all environments will depend on for deployment to work. Because the project config
+    # references this stack using !stack_output, all OTHER stacks in the project will depend upon
+    # this one.
+
+    # We override the project_code here so that there will only be ONE stack for deployment that is
+    # shared across all environments.
+    project_code: indigo-deployment
```

Comment:
isn't project_code in config/config.yaml overriding this since config/config.yaml depends on this template?

Reply:
I think you're thinking about stack group configs incorrectly. Stack group configs (such as config/config.yaml) don't actually "exist" in Sceptre. They don't depend on anything. Rather they are layers of key/value pairs that stacks in their groups _inherit_. Stack configs layer _on top of_ stack group configs.

In other words, if this Stack didn't override the project_code, it would inherit the project code from stack group in which it resides (config/project). Because config/project doesn't override it, then that value would be inherited from config/config.yaml. But this Stack _does_ override the project-code, so the project code used for this stack is indeed the one here.